lib/checkstl.cpp

```diff
@@ -701,8 +709,13 @@ static bool isSameIteratorContainerExpression(const Token* tok1,
     if (isSameExpression(true, false, tok1, tok2, library, false, false)) {
         return !astIsContainerOwned(tok1) || !isTemporary(true, tok1, &library);
     }
-    if (kind == ValueFlow::Value::LifetimeKind::Address) {
-        return isSameExpression(true, false, getAddressContainer(tok1), getAddressContainer(tok2), library, false, false);
+    if (kind == ValueFlow::Value::LifetimeKind::Address || kind == ValueFlow::Value::LifetimeKind::Iterator) {
+        const auto address1 = getAddressContainer(tok1);
+        const auto address2 = getAddressContainer(tok2);
+        for (std::size_t i = 0; i < address1.size(); ++i)
+            for (std::size_t j = 0; j < address2.size(); ++j)
+                if (isSameExpression(true, false, address1[i], address2[j], library, false, false))
+                    return true;
```

Comment:
I am not sure the reason for doing this, but it seems like it would be better to check if the entire set matches. This could be written as a nested `std::any_of` but it should probably be a nested `std::all_of`.

Reply:
It's for #12377. We used to get stuck on `pv`, which actually has two values (`get1` or `get2`).